services/wireguard/endpoint/kernelspace/client.go

```diff
@@ -83,6 +85,22 @@ func (c *client) AddPeer(iface string, peer wg.PeerInfo) error {
 	return c.wgClient.ConfigureDevice(iface, deviceConfig)
 }
 
+func (c *client) PeerStats() (consumer.SessionStatistics, error) {
+	d, err := c.wgClient.Device(c.iface)
+	if err != nil || d.Name != c.iface {
```

Comment:
Aren't we hiding an error here with the `d.Name != c.iface`? If the names don't match, all we'll get is a nil error and zero stats. Should we not at least log this occurrence?